### ISSUE TITLE
Creative Commons recommends against licensing software

### ISSUE THREAD

**Issue by moretrim** [BoyC/GW2TacO#2]
From the post announcing the source code release:

> The source code of TacO is released under the CC BY-NC 4.0 license.

Accordingly the release comes with a copy of the terms in the [`LICENSE`](https://github.com/BoyC/GW2TacO/blob/e4779330a095347a853f1c5d548f430d74521482/LICENSE) file. Unfortunately the makers of the CC licenses recommend against using them for software (with the exception of CC0):

> #### Can I apply a Creative Commons license to software?
>
> We recommend against using Creative Commons licenses for software. […]

(From the [FAQ](https://creativecommons.org/faq/#can-i-apply-a-creative-commons-license-to-software), which goes out of its way to explain the reasoning behind the advice.)

Normally at this point I would suggest a license that is designed for software and has widespread use as a replacement—everything to avoid bringing in discussing the finer legal points of software licensing, which tends to sour the mood fast. Unfortunately, I don’t know of any such license that would act as a close counterpart to CC BY-NC (specifically the prohibition of commercial use). Instead, I would recommend staying proprietary by default as it can never hurt the rightholders: All Rights Reserved and all that.

Sorry to bring up such a non-technical topic. At the end of the day though, the earlier this is addressed (i.e. before any contributions) the less pain incurred.

**Comment by BoyC** [comment]
Yeah. I've been made aware of the issue. This license was chosen specifically based on the requirement that this codebase not be used for commercial purposes by a copyright lawyer. I get that this makes it not "open source" but "source available" and that this license may not be a good fit for software (although in this case it's for the source code, not the compiled software).
My stance on the subject can be summed up as "don't make any money off of it, otherwise it's fair game".

**Comment by AsherGlick** [comment]
Can I ask, is the issue that you never want anyone make any money contributing to code you wrote? Or is it that you dont want another group to start selling an "advanced taco" for money with new features the original does not have?

**Comment by BoyC** [comment]
> Can I ask, is the issue that you never want anyone make any money contributing to code you wrote? Or is it that you dont want another group to start selling an "advanced taco" for money with new features the original does not have?

The issue is the basecode on which TacO rests that is quite powerful in itself and is the cumulative work of about 10 years that I don't want to give out for commercial use. This is the best compromise we can see where the community isn't deprived of a valuable resource and I still retain some control over what the code can be used for. TacO was always meant to be a freely available resource and it is in this spirit that I'm sharing this code anyway. I understand that this is alien to many people in the opensource community, however in the in the Demoscene where I come from the "use it but not for profit" point of view is a natural and accepted way of sharing software, quite commonly under a CC license such as this one.

**Comment by AsherGlick** [comment]
Thanks for responding, I was unfamiliar with the demoscene before today but I think I have a gist of where you are coming from. Definitely the license you chose makes sense for artistic creations. In general in the opensource world it is common for non-developers to fund developers to implement features or bugfixes on behalf of the users who cannot do it themselves. Many popular open source projects use this model, such as Blender, Krita, Godot Engine, and even Linux itself. As a result of bigger projects following this model it is rarer to see a license such as the one you are using which prevents such models. That however does not mean the one you chose is incorrect, just that it is not something we tend to see.

I feel you have already done a lot of research yourself on which license to use so I will not tell you that you should definitely use something else, but I would recommend you to take a look at a license like the GPL, if you have not already, in case it is something that satisfies you. It is slightly different from what you currently have, in the sense that a developer could make money using your code, but would still enforce that your code and all its derivatives stay free of cost for any user while being a more familiar license to the open source community.

**Comment by tildebyte** [comment]
@BoyC Late to the discussion, but... IANAL, but I've worked in open source for many years. It sounds to me like what you really want is the GPL v3 https://www.gnu.org/licenses/quick-guide-gplv3.html

I was going to paste a summary here, but it's simply too complex and detailed an issue for that. I highly recommend reviewing the Quick Guide at that link to at least understand what v3 is and does.

**Comment by ericcotelnu** [comment]
I'm no pro as to licensing either, but yeah, GPL v3 immediately came to mind. Sure, someone can fork / copy your code tree, sell it, but the source - they have to share regardless, and no one can take away anyone's right to access said codebase.

As an example: you'll see this with routers for instance, if they use the Lnux kernel, they **have to** provide source code access, and they **have to** provide the full license text of whichever GPL version the original source is under.

GPL, BSD, Apache, are the 3 most common licenses software is licensed under. Just some suggestions. I don't think I have to say this, but choosing wisely is your best option, IMO. Take your time, but not too much time, like a spiritual uncle of mine used to say :).
